back/app/controllers/concerns/blocking_profanity.rb

```diff
@@ -0,0 +1,52 @@
+module BlockingProfanity
+  extend ActiveSupport::Concern
+
+  class ProfanityBlockedError < StandardError
+    attr_reader :blocked_words
+    def initialize blocked_words
+      super
+      @blocked_words = blocked_words
+    end
+  end
+
+  SUPPORTED_CLASS_ATTRS = {
+    Comment.name => [:body_multiloc],
+    Idea.name => [:title_multiloc, :body_multiloc, :location_description],
+    Initiative.name => [:title_multiloc, :body_multiloc, :location_description]
+  }.freeze
+
+  included do
+    rescue_from ProfanityBlockedError, with: :render_profanity_blocked
+  end
+
+  def verify_profanity object
+    return if !AppConfiguration.instance.feature_activated? 'blocking_profanity'
+    
+    blocked_words = []
+    service = ProfanityService.new
+    attrs = SUPPORTED_CLASS_ATTRS[object.class.name]
+    attrs&.each do |atr|
+      next if object[atr].blank?
+      values = if atr.to_s.ends_with? '_multiloc'
+        object[atr]
```

Comment:
Is it because the value of the `if` statement is used in the assignment, that makes it confusing? Would you consider this to be more readable?
```
values = nil
if atr.to_s.ends_with? '_multiloc'
  values = object[atr]
else
  values = { nil => object[atr] }
end
```